src/main/resources/conf/configuration.xsd

```diff
@@ -0,0 +1,371 @@
+    </xs:sequence>
+    <xs:attribute default="true" name="processDataFiles" type="xs:boolean"/>
+    <xs:attribute default="true" name="storeJsonLabels" type="xs:boolean"/>
+    <xs:attribute default="true" name="storeLabels" type="xs:boolean"/>
+  </xs:complexType>
+
+  <xs:complexType name="file_ref_type">
+    <xs:annotation>
+      <xs:documentation xml:lang="en">
+        Describe the prefix of the filename to replace with new content.
+
+        @replacePrefix: the string to find in the filename starting with the
+                        first character of the filename (no regex or globbing)
+        @with: the replacement content
+
+        Example:
+           filename = /home/username/pdsfiles/someproduct.xml
+           replacePrefix = /home/username/pdsfiles
+           with = https://public.com/pdsfiles/root
+
+           After this fileRef is used, filename would be:
+              https://public.com/pdsfiles/root/someproduct.xml
+      </xs:documentation>
+    </xs:annotation>
+    <xs:attribute name="replacePrefix" type="xs:normalizedString"
+                  use="required"/>
+    <xs:attribute name="with" type="xs:normalizedString" use="required"/>
+  </xs:complexType>
+
+  <xs:complexType name="files_type">
+    <xs:annotation>
+      <xs:documentation xml:lang="en">
+      </xs:documentation>
+    </xs:annotation>
+    <xs:sequence>
+      <xs:element maxOccurs="unbounded" name="manifest" type="xs:normalizedString"/>
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:complexType name="filter_type">
+    <xs:annotation>
+      <xs:documentation xml:lang="en">
+        A simple container to hold two lists of strings, One for inclusion and
+        the other for exclusion,
+      </xs:documentation>
+    </xs:annotation>
+    <xs:sequence>
+      <xs:element maxOccurs="unbounded" minOccurs="0"
+                  name="exclude" type="xs:normalizedString"/>
+      <xs:element maxOccurs="unbounded" minOccurs="0"
+                  name="include" type="xs:normalizedString"/>
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:complexType name="harvest_configuration_type">
+    <xs:annotation>
+      <xs:documentation xml:lang="en">
+        These are the basic options for the harvest configuration file.
+
+        @nodeName: the PDS node that this harvest run applies to
+
+        autogenFields: should not be used except in development testing
+        do: tells where and how to harvest PDS4 labels
+        fileInfo: option allowing filename prefixes to be replaced
+        productFilter: should not be used except in development testing
+        registry: define the server harvest should use
+        xpathMaps: allow constraints in the PDS4 label to control harvesting
+      </xs:documentation>
+    </xs:annotation>
+    <xs:all>
+      <xs:element minOccurs="0"
+                  name="autogenFields" type="autogen_fields_type"/>
+      <xs:element name="do" type="do_type"/>
```

Comment:
@al-niessner Do we really need this? I would prefer to minimize the number of backwards incompatible changes if we can?

But if you would really prefer we encapsulte this, can we change this to `data_config` or `load_config` or just `data` or `load`?